rosbag2_storage/src/rosbag2_storage/metadata_io.cpp

```diff
@@ -48,6 +48,7 @@ struct convert<rosbag2_storage::TopicMetadata>
     node["name"] = topic.name;
     node["type"] = topic.type;
     node["serialization_format"] = topic.serialization_format;
+    node["offered_qos_profiles"] = topic.offered_qos_profiles;
```

Comment:
Is the reverse required within `decode`?

Reply:
This PR only supports storing the QoS profiles, it doesn't support loading them from a bag yet - that will take a bit of extra consideration that I want to put in a separate PR

Reply:
The extra consideration being "how do we pass metadata version to the sub-structs?" We should expect the value if it's metadata v4+ but not try to load it for previous versions of the metadata, but right now that version value is not available in this decoding context